scipy/spatial/transform/rotation.pyx

```diff
@@ -1016,6 +1019,106 @@ cdef class Rotation(object):
         else:
             return cls(quat, normalize=False, copy=False)
 
+    @classmethod
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def from_mrp(cls, mrp):
+        """Initialize from Modified Rodrigues Parameters (MRPs).
+
+        MRPs are a 3 element attitude description commonly used in the aerospace
+        industry. It is similar to the rotation vector, except instead of scaling
+        the rotation vector by the angle of rotation, is it scaled by tan(theta/4).
+
+        MRPs have a singuarity at 360 degrees which can be easily avoided by ensuring
+        that the angle of rotation is always <= 180
+
+        Parameters
+        ----------
+        mrp : array_like, shape (N, 3) or (3,)
+            A single vector or a stack of vectors, where `mrp[i]` gives
+            the ith set of MRPs.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the rotations represented by input rotation
+            vectors.
+
+        References
+        ----------
+        .. [1] Shuster, M. D. "A Survery of Attitude Representations",
+               The Journal of Astronautical Sciences, Vol. 41, No.4, 1993,
+               pp. 475-476
+
+        Notes
+        -----
+
+        .. versionadded:: 1.6.0
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Initialize a single rotation:
+
+        >>> r = R.from_mrp([0, 0, 1])
+        >>> r.as_euler('xyz')
+        array([0.        , 0.        , 3.14159265])
+        >>> r.as_euler('xyz').shape
+        (3,)
+
+        Initialize multiple rotations in one object:
+
+        >>> r = R.from_mrp([
+        ... [0, 0, 1],
+        ... [1, 0, 0]])
+        >>> r.as_euler('xyz')
+        array([[0.        , 0.        , 3.14159265],
+               [3.14159265, 0.        , 0.        ]])
+        >>> r.as_euler('xyz').shape
+        (2, 3)
+
+        It is also possible to have a stack of a single rotation:
```

Comment:
I saw your consistency argument.

Not sure, I'm fine with either keeping or removing it.